### PR TITLE
[SPARK-26650][core] Demote noisy HBase-related log message.

### PR DESCRIPTION
**Pull request by vanzin**
Make it a debug message so that it doesn't show up in the vast
majority of cases, where HBase classes are not available.

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
It seems to be ideal to have HBase token provider decoupled with core (as we did for Kafka). The message is noisy (not just noisy but also false alarm) for end users who don't use HBase in their query, but the message may help to track down missing classpath issue for end users who intentionally use HBase in their query.

**Comment by vanzin** [comment]
This is the same behavior as 2.4:
https://github.com/apache/spark/blob/branch-2.4/core/src/main/scala/org/apache/spark/deploy/security/HBaseDelegationTokenProvider.scala#L74

Putting this in a separate module is not just something that should be as separate change, but also means we'd have an opinion about what version of HBase Spark is built against. The current approach is not ideal, but we really don't want to dictate HBase versions to people, like we do for other components.

**Comment by HeartSaVioR** [comment]
Yeah and previously I have to request customer to turn on debug log and rerun query to get such information, hence I proposed to make the log level be consistent. Please see #23418. 

This patch suggests to roll back the change: when we want to roll back the log level, adding comment that `it is intentionally set to debug` would help avoiding to fixing level back and forth.

> Putting this in a separate module is not just something that should be as separate change, but also means we'd have an opinion about what version of HBase Spark is built against.

Please correct me if I'm missing here: if my understanding is right, once current approach doesn't deal with HBase version, putting this in a separate module also doesn't need to deal with HBase version (we will still rely on reflection even in that case). End users just need to dynamically link the new package if they use HBase with Kerberos enabled.

**Comment by vanzin** [comment]
> previously I had to request customer to turn on debug log and rerun query

And now you get an exception that is a couple of pages long every time you start Spark against a secure cluster, whether you're using HBase or not. That's worse.

> putting this in a separate module also doesn't need to deal with HBase version

Then there's no point in putting this in a separate module. Unless you're saying this should be moved to HBase or something external that is not part of the Spark distribution at all.

**Comment by HeartSaVioR** [comment]
> And now you get an exception that is a couple of pages long every time you start Spark against a secure cluster, whether you're using HBase or not. That's worse.

In my case the query crashed after 7 days because HBase token was not renewed due to classpath issue but surprisingly further error messages make me hard to guess there was classpath issue on HBase token provider. Without enabling debug log level I wouldn't find the root reason.

But I also agree there're more users to leverage HDFS so my change affected more users to be annoyed. Agree to roll back the log level: let's leave the reason of log level set to debug so no one would try to promote it again.

> Unless you're saying this should be moved to HBase or something external that is not part of the Spark distribution at all.

To be honest, once Spark doesn't support HBase in main repo and token provider is dynamically loadable as of now (I guess it didn't), I'm not 100% sure this is the right way to have HBase token provider in Spark repo/distribution, with dealing with reflection because we don't know which HBase version end users will link.

So basically I agree that should not be a part of Spark distribution: HBase connector can have it as of Spark 3.0.0. Don't have idea how Spark community has been dealing with ecosystem though.

**Comment by vanzin** [comment]
> In my case the query crashed after 7 days because HBase token was...

It's possible to debug that even if you don't have this message. If you're expecting HBase tokens but you don't see any of the info messages that are printed when they're created, well you have a problem.

Yeah, users can disable INFO logs, but they can also disable WARN logs (and I've seen many do that, which makes debugging anything pretty interesting).

> So basically I agree that should not be a part of Spark distribution

I don't disagree with that, but the main issue in the past is that there wasn't really a home for this, since HBase didn't really have an "official" Spark connector where this code could even live. If that's solved, maybe we could move this over there (once we sort out SPARK-26772).

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Review by HeartSaVioR — APPROVED** [review]
LGTM. I agree my statement for moving HBase token provider out is beyond the PR and given we're trying to address it (made it pluggable, etc) so eventually we can sort it out.

**Comment by HeartSaVioR** [comment]
(BTW sorry to open broader topic against small change to steal your time.)

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by HyukjinKwon** [comment]
Oops, I missed that the tests were running against the last commit. The last commit was only about adding comments so should be fine.
